modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/util/BaseQueryContext.java

```diff
@@ -108,14 +118,16 @@
     private BaseQueryContext(
             FrameworkConfig cfg,
             Context parentCtx,
-            IgniteLogger log
+            IgniteLogger log,
+            List<SqlExtension> extensions
     ) {
         super(Contexts.chain(parentCtx, cfg.getContext()));
         
         // link frameworkConfig#context() to this.
         this.cfg = Frameworks.newConfigBuilder(cfg).context(this).build();
         
         this.log = log;
+        this.extensions = extensions.stream().collect(Collectors.toMap(SqlExtension::name, Function.identity()));
```

Comment:
Is the map creation necessary for each query? Can we create the map once at the SqlProcessor?

Reply:
fixed